File: swift/easy/141-linked-list-cycle.swift
func hasCycle(_ head: ListNode?) -> Bool {
    var slow = head
    var fast = slow

    while fast?.next != nil {
        slow = slow!.next
        fast = fast!.next!.next

        if slow?.val == fast?.val {
            return true
        }
    }

    return false
}

File: swift/easy/141.linked-list-cycle.java
public boolean hasCycle(ListNode head) {
    ListNode tail = head;

    while (tail != null && tail.next != null) {
        head = head.next;
        tail = tail.next.next;

        if (head == tail) {
            return true;
        }
    }

    return false;
}
